Update project file and shared schemes to allow both a Carthage build and Xcode workspace build.

darinkrauss opened the pull request.

darinkrauss committed:
Do not automatically create schemes

darinkrauss committed:
Explicitly specify project dependencies

darinkrauss committed:
Create one shared scheme to improve Carthage build times

darinkrauss committed:
Create watchOS target and scheme

ps2 commented on LoopKit.xcodeproj/xcshareddata/xcschemes/Shared watchOS.xcscheme:
```diff
@@ -14,9 +14,9 @@
-               BlueprintName = "LoopKitUI"
+               BlueprintIdentifier = "A9E6758022713F4700E25293"
+               BuildableName = "LoopKit.framework"
+               BlueprintName = "LoopKit watchOS"
```
The pattern in loop is with a dash. I.e. `LoopKit-watchOS`.  Not strictly necessary to change, but if we want to keep a pattern.

ps2 reviewed (approved):
LGTM, just one naming suggestion, if you agree.

darinkrauss committed:
Use consistent iOS and watchOS scheme and target name convention

darinkrauss commented:
@ps2 I updated all of the targets in the affected repos to use the dash rather than space before `watchOS` (eg. `LoopKit-watchOS`). I also updated the schemes to be `Shared` and `Shared-watchOS` to match.

darinkrauss commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create a repo and workspace for building Tidepool Loop](https://trello.com/c/or5FC9tm/589-create-a-repo-and-workspace-for-building-tidepool-loop)
